synthesizer/tests/tests/process/execute/external_struct.aleo

```diff
@@ -0,0 +1,118 @@
+/*
+randomness: 45791624
+cases:
+  - program: child.aleo
+    function: call_make_array
+    inputs: []
+  - program: child.aleo
+    function: make_s
+    inputs: []
+  - program: child.aleo
+    function: make_t
+    inputs: []
+  - program: child.aleo
+    function: call_make_outer
+    inputs: []
+  - program: child.aleo
+    function: pass_s
+    inputs: []
+  - program: child.aleo
+    function: pass_t
+    inputs: []
+  - program: child.aleo
+    function: cast_s
+    inputs: []
+  - program: child2.aleo
+    function: call_make_outer2
+    inputs: []
+*/
+
+program parent.aleo;
+
+struct S:
+    x as field;
+
+struct Outer:
+    x as field;
+    s as S;
+
+function make_s:
+    cast 12field into r0 as S;
+    output r0 as S.public;
+
+function make_outer:
+    input r0 as field.public;
+    cast r0 into r1 as S;
+    cast r0 r1 into r2 as Outer;
+    output r2 as Outer.public;
+
+function make_array:
+    input r0 as field.public;
+    cast r0 into r1 as S;
+    cast r1 r1 into r2 as [S; 2u32];
+    output r2 as [S; 2u32].public;
+
+function accept_s:
+    input r0 as S.public;
+    output r0.x as field.public;
+
+/////////////////////////////////////////////////
+
+import parent.aleo;
+
+program child.aleo;
+
+struct T:
+    x as field;
+
+struct OuterT:
+    x as field;
+    s as T;
+
+struct Outer2:
+    s as parent.aleo/S;
```

Comment:
Can we have a negative test with a non-extisting external struct which we expect to fail somewhere? As we don't check this at the parsing level: https://github.com/ProvableHQ/snarkVM/issues/2599